packages/app-js/src/index.tsx

```diff
@@ -3,57 +3,83 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { ApiProps } from '@polkadot/ui-api/types';
-import { AppProps, I18nProps } from '@polkadot/ui-app/types';
-import { Log, LogType } from './types';
+import { AppProps } from '@polkadot/ui-app/types';
+import { CustomWindow,Log, LogType } from './types';
 
 import React from 'react';
 import { withApi, withMulti } from '@polkadot/ui-api/index';
-import { Button } from '@polkadot/ui-app/index';
+import { Button, Dropdown } from '@polkadot/ui-app/index';
 import uiKeyring from '@polkadot/ui-keyring';
 import * as util from '@polkadot/util';
 import * as hashing from '@polkadot/util-crypto';
 
 import './index.css';
+import snippets from './snippets';
+import translate from './translate';
 
+import Intro from './Intro';
 import Editor from './Editor';
 import Output from './Output';
-import translate from './translate';
 
-type Props = ApiProps & AppProps & I18nProps;
+type Props = ApiProps & AppProps;
 type State = {
   code: string,
-  logs: Array<Log>
+  logs: Array<Log>,
+  snippet: string
 };
 
+const customWindow: CustomWindow = window;
+
+const local = snippets.find(obj => obj.value === localStorage.getItem('app-js-snippet'));
+
 class App extends React.PureComponent<Props, State> {
   state: State = {
-    code: '',
-    logs: []
+    code: local ? local.code : snippets[0].code,
+    logs: [],
+    snippet: local ? local.value : snippets[0].value
   };
 
   render () {
-    const { logs } = this.state;
+    customWindow.api = this.props.api;
+    customWindow.keyring = uiKeyring;
+    customWindow.utilCrypto = hashing;
+    customWindow.util = util;
+
+    const { code, logs, snippet } = this.state;
 
     return (
       <main className='js--App'>
-        <Editor onEdit={this.onEdit}>
-          <Button
-            className='action-button'
-            isCircular
-            isPrimary
-            icon='play'
-            onClick={this.runJs}
+        <header className='container'>
+          <Intro />
+          <Dropdown
+            className='js--Dropdown'
+            onChange={(value) => this.selectExample(value)}
+            options={snippets.map(({ code, ...options }) => ({ ...options }))}
```

Comment:
As per above, probably better calculating these once and keeping in the state.

Reply:
Since they're static and not getting updated (as of now) i just moved the declaration from the options in  return() to the render function. 
If we decide to save  the edited code as a dropdown option I'd move it to the state.